modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/DropSchemaCommand.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands;
+
+import static org.apache.ignite.internal.catalog.commands.CatalogUtils.schemaOrThrow;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.CatalogCommand;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSchemaDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor;
+import org.apache.ignite.internal.catalog.storage.DropIndexEntry;
+import org.apache.ignite.internal.catalog.storage.DropSchemaEntry;
+import org.apache.ignite.internal.catalog.storage.DropTableEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+
+/**
+ * A command that drops a schema with specified name.
+ */
+public class DropSchemaCommand implements CatalogCommand {
+    /** Returns builder to create a command to drop schema with specified name. */
+    public static DropSchemaCommandBuilder builder() {
+        return new Builder();
+    }
+
+    private final String schemaName;
+
+    private final boolean cascade;
+
+    /**
+     * Constructor.
+     *
+     * @param schemaName Name of the schema.
+     * @param cascade Flag indicating forced deletion of a non-empty schema.
+     * @throws CatalogValidationException if any of restrictions above is violated.
+     */
+    private DropSchemaCommand(String schemaName, boolean cascade) throws CatalogValidationException {
+        this.schemaName = schemaName;
+        this.cascade = cascade;
+    }
+
+    @Override
+    public List<UpdateEntry> get(Catalog catalog) {
+        if (CatalogUtils.isSystemSchema(schemaName)) {
+            throw new CatalogValidationException("System schema can't be dropped [name={}].", schemaName);
+        }
+
+        CatalogSchemaDescriptor schema = schemaOrThrow(catalog, schemaName);
+
+        if (!cascade && (schema.indexes().length > 0 || schema.tables().length > 0)) {
```

Comment:
does it make sense to introduce `isEmpty()` method on schema object?

Reply:
Done :thinking: 
```
    /** Returns {@code true} if the schema contains doesn't contain any indexes or tables, otherwise {@code false}. */
    public boolean isEmpty() {
        return tables.length == 0 && indexes.length == 0;
    }
```